src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -111,37 +114,26 @@ def to_json_dict(self) -> dict:
             "artifact": self.artifact,
             "version": self.version,
             "packaging": self.packaging,
+            "classifier": self.classifier,
         }
         return ret
 
     @classmethod
-    def from_coord_str(cls, s: str, *, with_2315_workaround=False) -> Coordinate:
-        """Parses from a coordinate string with an optional `packaging` string.
+    def from_coord_str(cls, s: str) -> Coordinate:
+        """Parses from a coordinate string with optional `packaging` and `classifier` coordinates.
 
-        ${organisation}:${artifact}:${version}[:${packaging}]
-
-        NB: The `with_2315_workaround` flag is for parsing in locations that are affected by
-          https://github.com/coursier/coursier/issues/2315
-        ... which swaps the `version` and `packaging`.
+        ${organisation}:${artifact}[:${packaging}[:${classifier}]]:${version}
         """
-        parts = s.split(":")
-
-        if len(parts) == 4:
-            if with_2315_workaround:
-                version = parts[3]
-                packaging = parts[2]
-            else:
-                version = parts[2]
-                packaging = parts[3]
-        else:
-            version = parts[2]
-            packaging = "jar"
-
+        regex = re.compile("([^: ]+):([^: ]+)(:([^: ]*)(:([^: ]+))?)?:([^: ]+)")
```

Comment:
Can you move the compilation of this regex out into a constant on the class? I know that regex compilation has caches, but have been bitten before.